qwandaq/src/main/java/life/genny/qwandaq/entity/SearchEntity.java

```diff
@@ -15,46 +17,52 @@
 import life.genny.qwandaq.attribute.EntityAttribute;
 import life.genny.qwandaq.datatype.DataType;
 import life.genny.qwandaq.entity.search.clause.And;
+import life.genny.qwandaq.entity.search.clause.ClauseArgument;
 import life.genny.qwandaq.entity.search.clause.ClauseContainer;
 import life.genny.qwandaq.entity.search.clause.Or;
+
 import life.genny.qwandaq.entity.search.trait.Action;
-import life.genny.qwandaq.entity.search.trait.AssociatedColumn;
 import life.genny.qwandaq.entity.search.trait.Column;
 import life.genny.qwandaq.entity.search.trait.Filter;
 import life.genny.qwandaq.entity.search.trait.Sort;
+import life.genny.qwandaq.entity.search.trait.Trait;
 
 /* 
  * SearchEntity class implements the search of base entities applying 
  * different filters/search to the baseEntity and its attributes.
  */
+@SuppressWarnings("unchecked")
 @RegisterForReflection
 public class SearchEntity extends BaseEntity {
 
 	private static final Logger log = Logger.getLogger(SearchEntity.class);
 
 	private static final long serialVersionUID = 1L;
 
+	// TODO: Polish this
+	private Map<Class<? extends Trait>, List<? extends Trait>> traits = new HashMap<>();
```

Comment:
Does this map retain the indexed order that sorts and columns, etc. were added in?

Reply:
It does, given it stores a list as a value the order will be retained